tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2966,84 +2967,134 @@ llvm::Optional<Value> convertResizeOp(PatternRewriter& rewriter, Operation* op,
   size_t output_height = output_shape[1];
   size_t output_width = output_shape[2];
 
-  double fp_stride_y =
-      static_cast<double>(input_height) / static_cast<double>(output_height);
-  double fp_stride_x =
-      static_cast<double>(input_width) / static_cast<double>(output_width);
-  if (align_corners && output_height > 1) {
-    fp_stride_y = static_cast<double>(input_height - 1) /
-                  static_cast<double>(output_height - 1);
-  }
-  if (align_corners && output_width > 1) {
-    fp_stride_x = static_cast<double>(input_width - 1) /
-                  static_cast<double>(output_width - 1);
-  }
-
-  double fp_offset_y, fp_offset_x;
-  if (half_pixel_centers) {
-    fp_offset_y = fp_stride_y * 0.5f - 0.5f;
-    fp_offset_x = fp_stride_x * 0.5f - 0.5f;
-  } else {
-    fp_offset_y = 0.0f;
-    fp_offset_x = 0.0f;
-  }
+  auto isPowerOfTwoUpscale = [](size_t a, size_t b) {
+    auto isPowerOfTwo = [](size_t x) { return x && (!(x & (x - 1))); };
+    return (a % b == 0) && isPowerOfTwo(a / b);
+  };
 
-  // oh * fp_stride_y + fp_offset_y = ix
+  auto gcd = [](size_t a, size_t b) {
```

Comment:
Could llvm::GreatestCommonDivisor64 be used here?

Reply:
Cool. That's what I was looking for! I haven't noticed this utility before. Thanks.